app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
+    people: dict = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    spouse_types: set[str] = {"wife", "husband"}
+    for person in people:
+        if person["name"] in Person.people:
+            continue
+        current_person = Person(person["name"], person["age"])
+        Person.people[person["name"]] = current_person
+        spouse_type: set = (spouse_types & person.keys())
+        if len(spouse_type) < 1 or person.get(next(iter(spouse_type))) is None:
```

Comment:
u can use pop method instead of iter + next to write less code ) the same is for other cases

Reply:
I disagree, that will remove an item from my template set. I don't want to do that

Reply:
u can make in 16th line like this 
`spouse_type: str = (spouse_types & person.keys()).pop()`
then u can use this variable in ur code 
in 29th line u can use `remove` method instead of `-`
this will make u code DRY - cause now u are making the same thing every time u are needed of spouse_type
why don't u create a variable for this ?

Reply:
that's a valid point, but semantically seems a bit weird. Looks like python is still chinese to me :(